### PR TITLE
211 implement main menu

### PR DESCRIPTION
**Pull request by Sean-Nishi**

### CODE REVIEW

File: game/game.py [hunk 1]
```diff
@@ -31,23 +31,27 @@ def __init__(self):
...
                     sys.exit()
 
-            # run level
-            self.level.run()
+            self._quit_game = self._level_manager.run()
 
             # update display based on events
             pygame.display.update()
 
+        # if we quit, exit game
```
Comment (owalk) [inline game/game.py]: remove this comment so its not added and removed in same PR

File: game/levels/level.py [hunk 2]
```diff
@@ -148,6 +148,20 @@ def create_tile_group(self, layout: List[int]) -> pygame.sprite.Group:
 
         return sprite_group
 
+    def add_obstacles(self):
```
Comment (owalk) [inline game/levels/level.py:186]: add return type hint

File: game/levels/levelManager.py [hunk 3]
```diff
@@ -26,23 +27,40 @@ def __init__(self, surface: pygame.Surface) -> None:
         gameData: str
             Path to gameData.py. Parse file for specific level info. MainManu is default
         """
+        self._screen = pygame.display.get_surface()
+        self.clock = pygame.time.Clock()
```
Comment (owalk) [inline game/levels/levelManager.py]: Either todo for added setter/getter to clock, or switch to setter/getter

File: game/levels/mainMenu.py [hunk 4]
```diff
@@ -1,119 +1,76 @@
...
     """Main Menu Level.
 
-    This class will manage and load all sprites for the main menu level
+    This class will manage and load all sprites for the main menu level.
+
+    Attributes
+    ----------
+    _menu_image: pygame.Surface
+        The background image for the main menu
+
```
Comment (owalk) [inline game/levels/mainMenu.py]: rm newline

File: game/levels/mainMenu.py [hunk 5]
```diff
@@ -1,119 +1,76 @@
...
-        # load menu image
-        self.menu_image = pygame.image.load(MAIN_MENU_BACKGROUND_PATH)
-        # font
-        self.font = pygame.font.SysFont("Corbel", 40)
-        # font color, black
-        self.font_color = (255, 255, 255)
+        super().__init__("Main Menu", WINDOW_WIDTH, WINDOW_HEIGHT)
+        self._menu_image: pygame.Surface = pygame.image.load(MAIN_MENU_BACKGROUND_PATH)
+        self._display_surface = pygame.display.get_surface()
+        # create a pygame_menu inside our display surface
```
Comment (owalk) [inline game/levels/mainMenu.py:46]: update comment

File: game/levels/mainMenu.py [hunk 6]
```diff
@@ -1,119 +1,76 @@
...
+        self.add_menu_options()
+        # TODO: self._user_selection should be Enum
+        self._user_selection: str = "None"
+
+    def set_user_selection(self, selection: str):
+        """Set user selection for LevelManager.
```
Comment (owalk) [inline game/levels/mainMenu.py:52]:
```Suggestion
        """Set user selection for MainMenu.

        Intended for use in LevelManager.
        """
```

File: game/levels/mainMenu.py [hunk 7]
```diff
@@ -1,119 +1,76 @@
...
-        self.font_color = (255, 255, 255)
+        super().__init__("Main Menu", WINDOW_WIDTH, WINDOW_HEIGHT)
+        self._menu_image: pygame.Surface = pygame.image.load(MAIN_MENU_BACKGROUND_PATH)
+        self._display_surface = pygame.display.get_surface()
+        # create a pygame_menu inside our display surface
+        self.add_menu_options()
+        # TODO: self._user_selection should be Enum
+        self._user_selection: str = "None"
+
+    def set_user_selection(self, selection: str):
```
Comment (owalk) [inline game/levels/mainMenu.py:51]: add return type hint

File: game/cameraControl/cameraManager.py [hunk 8]
```diff
@@ -12,16 +12,16 @@ class CameraManager(pygame.sprite.Group):
...
-    displaySurface: Surface
+    _displaySurface: pygame.Surface
         The game surface in which to render all sprites
-    offset: vector2
-        The offset at which to render all sprites
-    halWidth: int
+    _halfWidth: int
         Half the display surface width
-    halfHeight: int
+    _halfHeight: int
```
Comment (owalk) [inline game/cameraControl/cameraManager.py:19]: snake case

File: game/cameraControl/cameraManager.py [hunk 9]
```diff
@@ -12,16 +12,16 @@ class CameraManager(pygame.sprite.Group):
...
     ----------
-    playerCharacter: Player
-        The currently shown frame represented by an index
-    displaySurface: Surface
+    _displaySurface: pygame.Surface
         The game surface in which to render all sprites
-    offset: vector2
-        The offset at which to render all sprites
-    halWidth: int
+    _halfWidth: int
```
Comment (owalk) [inline game/cameraControl/cameraManager.py:17]: snake case

File: game/cameraControl/cameraManager.py [hunk 10]
```diff
@@ -12,16 +12,16 @@ class CameraManager(pygame.sprite.Group):
 
     Attributes
     ----------
-    playerCharacter: Player
-        The currently shown frame represented by an index
-    displaySurface: Surface
+    _displaySurface: pygame.Surface
```
Comment (owalk) [inline game/cameraControl/cameraManager.py:15]: snake case

File: game/cameraControl/cameraManager.py [hunk 11]
```diff
@@ -12,16 +12,16 @@ class CameraManager(pygame.sprite.Group):
...
         The game surface in which to render all sprites
-    offset: vector2
-        The offset at which to render all sprites
-    halWidth: int
+    _halfWidth: int
         Half the display surface width
-    halfHeight: int
+    _halfHeight: int
         Half the display surface height
+    _offset: vector2
```
Comment (owalk) [inline game/cameraControl/cameraManager.py:21]: snake case

File: game/cameraControl/cameraManager.py [hunk 12]
```diff
@@ -12,16 +12,16 @@ class CameraManager(pygame.sprite.Group):
...
+        The offset at which to render all sprites
+    _playerCharacter: Player
```
Comment (owalk) [inline game/cameraControl/cameraManager.py:23]: snake case

File: game/cameraControl/cameraManager.py [hunk 13]
```diff
@@ -38,7 +38,7 @@ def __init__(self, playerCharacter: Player) -> None:
         Parameters
         ----------
         playerCharacter: Player
-            The player character
+            The player character entities move around
```
Comment (owalk) [inline game/cameraControl/cameraManager.py:41]:
```Suggestion
            The player character that entities move around
```

File: game/game.py [hunk 14]
```diff
@@ -44,11 +63,7 @@ def run(self):
                     sys.exit()
 
             self._quit_game = self._level_manager.run()
-
-            # update display based on events
             pygame.display.update()
-
-        # if we quit, exit game
```
Comment (owalk) [inline game/game.py]: should not rlly get added in prev commit in the same PR

File: game/levels/level.py [hunk 15]
```diff
@@ -16,6 +16,38 @@ class Level(object):
...
     The Level class will instantiate all objects required for a single level to run.
+
+    Attributes
+    ----------
+    _display_surface: pygame.Surface
+        The surface which to display the level
+    _universal_assets: List[pygame.sprites]
+        The list of all art assets from LevelManager
+    _mixer: pygame.mixer
+        LevelManager's music mixer which to load level specific music to
```
Comment (owalk) [inline game/levels/level.py:27]:
```Suggestion
        LevelManager's music mixer for level specific music
```

File: game/levels/mainMenu.py [hunk 16]
```diff
@@ -7,17 +7,32 @@
...
     Methods
     -------
+    get_user_selection(self)
+        Get the user selection
     set_user_selection(self, selection: str)
+        Sets the _user_selection
+    add_menu_options(self, selection: str)
+        Create all buttons and selectors for the menu
+    start_game(self)
+        Changes user selection to "Level" and disables the menu
```
Comment (owalk) [inline game/levels/mainMenu.py:31]:
```Suggestion
        Changes user selection to Level instance and disables the menu
```

**Review by owalk — CHANGES_REQUESTED** [review]

File: game/cameraControl/cameraManager.py (hunk 13 above)
Comment (Sean-Nishi) [reply game/cameraControl/cameraManager.py:45]: Also changed init parameter and docstring to snake_case to be consistent

Comment (owalk) [reply game/cameraControl/cameraManager.py:45]: This comment was missed

File: game/levels/level.py (hunk 15 above)
Comment (owalk) [reply game/levels/level.py:27]: this comment was missed

File: game/levels/mainMenu.py (hunk 16 above)
Comment (owalk) [reply game/levels/mainMenu.py:31]: This comment was missed

**Comment by owalk** [comment]
@Sean-Nishi Can you assign yourself to the PR and include any issues this will close

**Review by owalk — APPROVED** [review]

**Pull request merged**